Add deprecation notices (#1229)
kyma-project/serverless: components/serverless/design/scaling_functions.md

However, there are some implementation conflicts between the two features. This is a design and an implementation plan to unify the UX while using Functions as scaled resources.

> [!CAUTION]

> > The `spec.ScaleConfig` field will be removed in a future serverless version, users will have to create HPA manually

Reviewer: ```suggestion
> The `spec.ScaleConfig` field will be removed in a future Serverless version, where you must create HPA manually
```
